Rem/Bot/BotState.cs

```diff
@@ -32,16 +30,16 @@ public static BotState Initialize(string version, string filePath)
             {
                 var state = JsonConvert.DeserializeObject<BotState>(File.ReadAllText(filePath));
                 state.FilePath = filePath;
-                if (state.AdminList == null)
-                {
-                    state.AdminList = new HashSet<ulong>();
-                }
+                state.AdminList = state.AdminList ?? new HashSet<ulong>();
+
                 Task.WaitAll(state.PersistState());
                 state.Version = version;
                 return state;
             }
-            catch (Exception)
+            //TODO: Handling of exception needed.
+            catch (ArgumentException e)
```

Comment:
Exception is to catch if a file is not found. We cannot simply check for file existence because it is a possible race condition (file may be created in between the check and read).

Reply:
ArgumentException will not catch this.

Reply:
not used to non-javadocs so I saw the Arg exceptions thrown in `ReadAllText()` and thought that was genericized enough, but didn't look into the `InternalRead()` method. Usually catching 'Exception' is frowned upon because it's harder for another developer to understand what can break.

Reply:
In this case it's fine because we want to catch *any* case where reading settings fails.